### PR TITLE
Chemistry test improvements [#176971735]

### PR DESCRIPTION
**Pull request by kswenson**
- air/water temperature tests show images and results when appropriate
- turbidity test shows images when appropriate
- infrastructure in place for all tests to show images
- infrastructure in place for all tests to specify animations
- air/water temperature tests specify their ultimate animations
- turbidity test specifies its ultimate animation

[[#176971735]](https://www.pivotaltracker.com/story/show/176971735)
Testable at https://leaf-pack.concord.org/branch/temperature/.

@mklewandowski I figured out how to represent animations internally as well as how to represent individual images in lieu of animations. You'll see that in the air/water temperature tests and the turbidity test. I did not end up doing any investigation into the animation itself, although I am confident that with these representations in place implementing the animations shouldn't be too difficult.

### CODE REVIEW

File: src/utils/chem-utils.ts [hunk 1]
```diff
@@ -97,6 +220,42 @@ export const chemistryTests: ChemistryTest[] = [
...
+  },
+  [EnvironmentType.environment4]: {
+    [ChemTestType.airTemperature]: 38,
+    [ChemTestType.waterTemperature]: 24,
+    [ChemTestType.pH]: 9,
+    [ChemTestType.nitrate]: 20,
+    [ChemTestType.turbidity]: 100,
+    [ChemTestType.dissolvedOxygen]: 4
+  }
+};
```
Comment (mklewandowski) [inline src/utils/chem-utils.ts:257]: So these are the values from the scenario that get fed to the chem lab?  Looks good.

File: src/utils/chem-utils.ts [hunk 2]
```diff
@@ -85,9 +183,34 @@ export const chemistryTests: ChemistryTest[] = [
...
+      {type: StepType.animation, label: t("CHEM.COLLECT.SAMPLE"), Image: "1f",
+        frames: [
+          {label: "0", image: Turbidity0, duration: 0.25},
+          {label: "1a", image: Turbidity1A, duration: 1},
+          {label: "1b", image: Turbidity1B, duration: 2},
+          {label: "1c", image: Turbidity1C, duration: 0},
+          {label: "1d", image: Turbidity1D, duration: 2},
+          {label: "1d-", image: "none", duration: 0},
+          {label: "1e", image: "byValue", duration: 1},
+          {label: "1f", image: "byValue", duration: -1},
```
Comment (mklewandowski) [inline src/utils/chem-utils.ts:197]: Did we check to see if there are cases where we need to store a fade time as well?

Comment (kswenson) [reply src/utils/chem-utils.ts:197]: I didn't look through all the tests, but there were no other fade times in the tests I looked at.

Comment (mklewandowski) [reply src/utils/chem-utils.ts:197]: I'm mostly thinking of the case where we mix the sample.  I don't see it in the spec, but it clearly uses different fade timing in the proto:
mix

**Review by mklewandowski — APPROVED** [review]
👍 all of this is looking good.  I'll need to look more closely on Monday, but I don't want to hold anything up.  My main concern is about the fades times.  From looking at the proto, there must be cases where we have different fade times.

File: src/utils/chem-utils.ts (hunk 2 above)
Comment (kswenson) [reply src/utils/chem-utils.ts:197]: I just looked and the spec has them all at 0.5 sec so I figure we start there and if we have to make it more elaborate we can do so.

Comment (mklewandowski) [reply src/utils/chem-utils.ts:197]: Sounds good.

**Pull request merged**